docs/content/best-practices/modeling.mdx

```diff
@@ -0,0 +1,176 @@
+
+## Custom Roles
+
+However, in some applications, you will want end-users to define their own roles. In that case, you do need a role type:
+
+```dsl.openfga
+ model
+    schema 1.1 
+ type user
+
+  type role
+   relations
+     define assignee: [user]
+
+  type entity
+    relations
+      define admin: [user]
+      define billing_manager: [user]
+
+      define can_manage_billing: [role#assignee] or admin or billing_manager
+      define can_manage_users: [role#assignee] or admin
+```
+
+In this example we are combining static roles with dynamic roles. This is the recommended way of doing it. Do not always use generic roles. Define static roles for the ones you already know, and implement generic roles if your application needs them. Adding additional static roles is very easy, you just need to add a relation in the model, and it does not happen often. 
+
+You can see a full example of implementing custom roles [here](https://github.com/openfga/sample-stores/tree/main/stores/custom-roles).
+
+## Organizational Hierarchies
+
+If you are building a B2B SaaS application, you might encounter the following requirements:
+
+- You need the employees of your organization to access customer data for help desk scenarios, or as super-admin
+- Your customers have a hierarchical organization structure that you'll need to represent. 
+
+A possible way of modeling this would be with a recursive entity type:
+
+```dsl.openfga
+ model
+    schema 1.1 
+
+ type user
+ type entity
+    relations
+      define parent: [entity]
+      define admin: [user] or admin from parent
+      define member: [user] or member from parent
+```
+
+You can use this construct to solve both problems. You define a root entity to represent your company, and then each entity can have a set of child entities that will inherit permissions. However, it always has a recursive relation that is more expensive to evaluate, and it does not fully represent the two different hierarchies you need.
+
+If have the requirement of allowing the employees of your company to access the system, we recommend to use the model below.
+
+```dsl.openfga
+model 
+  schema 1.1
+
+ type user
+ type system
+    relations
+       define admin: [user] 
+
+ type entity
+    relations
+      define admin: [user] or admin from system
+      define member: [user] 
+```
+This defines a hierarchy that is not recursive, which is faster to evaluate, and is easier to understand the model intent. You can see a full example of this scenario [here](https://github.com/openfga/sample-stores/blob/main/stores/superadmin).
+
+If you also have the requirement of supporting hierarchical organizations, you can add that when you need it:
+
+```dsl.openfga
+model 
+  schema 1.1
+
+ type user
+ type system
+    relations
+       define admin: [user] 
+
+ type entity
```

Comment:
Is there a reason we're calling these entities instead of orgs here (and entities instead of applications above)?